worker/uniter/runner/debug/server.go

```diff
@@ -40,6 +44,9 @@ func (s *ServerSession) RunHook(hookName, charmDir string, env []string) error {
 	cmd.Env = env
 	cmd.Dir = charmDir
 	cmd.Stdin = bytes.NewBufferString(debugHooksServerScript)
+	var combinedOutput bytes.Buffer
```

Comment:
This means a hook's output won't show up until it completes, which can take an arbitrarily long amount of time. This may have an impact on usability of debug-hooks.

Please either add Stdout and Stderr parameters to HookContext/ServerSession and use them, or modify echocommand to do what you need.